### PR TITLE
Added Subscriptions and Users Routes

### PR DESCRIPTION
**Pull request by henrybv**

### CODE REVIEW

File: server/app/routes/users/index.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+module.exports = router;
+
+router.get('/', (req, res, next) => {
+	Users.find({})
+		.then(users => res.json(users))
+		.catch(next);
+});
+
+router.post('/', (req, res, next) => {
+	Users.findOrCreate(req.body)
```
Comment (jisaf) [inline server/app/routes/users/index.js:17]: mongoose doesn't have a findOrCreate, we either need a static to do this or a module

File: server/app/routes/users/index.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		.then(newUser => res.json(newUser))
+		.catch(next);
+});
+
+router.params('/:userId', (req, res, next, id) => {
+	Users.findById(id)
+		.populate('movieQueue')
```
Comment (danetomseth) [inline server/app/routes/users/index.js:24]: also need to populate address, subscription, and billinghistory

File: server/app/routes/users/index.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+				next();
+			}
+		})
+		.catch(next);
+
+});
+
+router.get('/:userId', (req, res, next) => {
+	res.json(req.user)
+});
```
Comment (jisaf) [inline server/app/routes/users/index.js]: Catch

File: server/app/routes/users/index.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+router.get('/:userId/moviequeue', (req, res, next) => {
+	res.json(req.user.movieQueue);
+});
```
Comment (jisaf) [inline server/app/routes/users/index.js]: Catch

File: server/app/routes/users/index.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+router.params('/:userId', (req, res, next, id) => {
+	Users.findById(id)
+		.populate('movieQueue')
+		.populate('address')
+		.populate('billingHistory')
+		.then(user => {
+			if(!user) {
+				res.sendStatus(404);
+			} else {
+				req.user = user;
```
Comment (jisaf) [inline server/app/routes/users/index.js]: FSG uses req.user for login and sessions
maybe use req.modelUser instead

**Pull request merged**